### PR TITLE
fix cell background color on default and selected state

### PR DESCRIPTION
**Pull request by LaserPatrick**

### CODE REVIEW

File: Segmentio/Source/Cells/SegmentioCell.swift [hunk 1]
```diff
@@ -139,6 +139,8 @@ class SegmentioCell: UICollectionViewCell {
         if style.isWithText() {
             segmentTitleLabel?.textColor = selected ? selectedState.titleTextColor : defaultState.titleTextColor
             segmentTitleLabel?.font = selected ? selectedState.titleFont : defaultState.titleFont
+            backgroundColor = selected ? selectedState.backgroundColor : defaultState.backgroundColor
+            segmentImageView?.tintColor = selected ? selectedState.imageTintColor : defaultState.imageTintColor
```
Comment (rrolland) [inline Segmentio/Source/Cells/SegmentioCell.swift]: @LaserPatrick In order to also manage style .onlyImage, you can move this code after the if statement.

**Comment by LaserPatrick** [comment]
edited

**Comment by Tykhonkov** [comment]
@LaserPatrick  can you provide more information about PR, and what issue it's suppose to fix?

**Comment by LaserPatrick** [comment]
Hi @Tykhonkov Sure, my initial PR was due to the selectedState.backgroundColor was never used and then I applied it to the cell backgroundColor. Second I added the imageTintColor status to set the tint of the image if you use the templated image.

Finaly I removed the Void param to support swift 4.0 on xcode 9.0

**Comment by Tykhonkov** [comment]
I have closed it because the color of cell shouldn't be changed.
This screen helps to understand why. 
Preview

for changing background color you should use ```SegmentioOptions```

Preview

Image tint color is a good thing, but it should be changed only when animation of selection state is completed.

Swift 4 migration had been already done in another PR.

**Pull request closed without merging**